src/App.tsx

```diff
@@ -1,26 +1,73 @@
+/* eslint-disable react-hooks/exhaustive-deps */
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import { Loader, TodoFilter, TodoList, TodoModal } from './components';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { useEffect, useState } from 'react';
+import { todosSlice } from './features/todos';
+import { getTodos } from './api';
+import { StatusType } from './features/filter';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+export const App = () => {
+  const { setTodos } = todosSlice.actions;
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+  const dispatch = useAppDispatch();
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const todos = useAppSelector(state => state.todos);
+  const filter = useAppSelector(state => state.filter);
+
+  const [isLoad, setIsLoad] = useState(false);
+
+  useEffect(() => {
+    setIsLoad(true);
+
+    getTodos()
+      .then(response => {
+        let queried = response.filter(todo =>
+          todo.title.toLowerCase().includes(filter.query.toLowerCase()),
+        );
+
+        switch (filter.status) {
+          case StatusType.ACTIVE: {
+            queried = queried.filter(todo => !todo.completed);
+
+            break;
+          }
+
+          case StatusType.COMPLETED: {
+            queried = queried.filter(todo => todo.completed);
+
+            break;
+          }
+        }
+
+        dispatch(setTodos(queried));
```

Comment:
I think it is better to store all todos in state and do filtering on client side if filtering is not supported on BE side

Reply:
It was pointed out in the task condition

Reply:
After all in this case Loader isn't appropriate because react makes calculations instantly. I agree with you that it's better in general because it requires resources to ask data from server

Reply:
I have added a var turn on Loader if data loads